src/radical/pilot/agent/agent_0.py

```diff
@@ -340,77 +353,130 @@ def _start_services(self):
         If a `./services` file exist, reserve a compute node and run that file
         there as bash script.
         '''
-
-        if not os.path.isfile('./services'):
-            return
+        service_descriptions = self._cfg.services
+        services = list()
+        self._service_task_ids = {}
```

Comment:
If possible, please declare class members in the constructor.  Otherwise you imply a certain sequence of method calls which might not always be given (remember that the state callback is being called asynchronously).

Reply:
moved